core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -299,26 +329,32 @@ class LoadBalancerService(config: WhiskConfig, entityStore: EntityStore)(implici
      * Invoker is currently using and which is better avoid if/until
      * these are moved to some common place (like a subclass of Message?)
      */
-    private val activationCountMap = TrieMap[(String, String), AtomicInteger]()
-    private def hashAndCountSubjectAction(msg: ActivationMessage): (Int, Int) = {
+    private def hashSubjectAction(msg: ActivationMessage): Int = {
         val subject = msg.user.subject.asString
         val path = msg.action.toString
-        val hash = subject.hashCode() ^ path.hashCode()
-        val key = (subject, path)
-        val count = activationCountMap.get(key) match {
-            case Some(counter) => counter.getAndIncrement()
-            case None => {
-                activationCountMap.put(key, new AtomicInteger(0))
-                0
-            }
-        }
-        return (hash, count)
+        subject.hashCode() ^ path.hashCode()
+    }
+
+    /** Memoizes the result of `f` for later use. */
+    private def memoize[I, O](f: I => O): I => O = new scala.collection.mutable.HashMap[I, O]() {
```

Comment:
Agree on the first, disagree on the latter: Don't think we need to be so cautious with this particular function. At steady state the results shouldn't change much. Folding it into the invokerHealth protocol won't guarantee no overlapping recomputation either. We could mitigate this by storing a Promise's future in the map and return that to each caller so we would fold all computations into one.